### ISSUE TITLE
Using a trajectory from .log file

### ISSUE THREAD

**Issue by argosvr** [IntelVCL/Open3D#421]
Hello,
In this document http://www.open3d.org/docs/tutorial/Advanced/rgbd_integration.html you refer to "Read trajectory from .log file".
But I can't find something in the documentation which refers to "create a trajectory from .log file".
Is it possible to generate such a log using the RGDB odometry ?

**Comment by syncle** [comment]
Hi @argosvr. It's good question. Currently Open3D supports full interface with .json file format because it can contain multiple attributes, such as confidence, odometry, information matrix, source, target index and so on. 

.log file example is just provided for convenience. There is a python I/O script for .log file in src/Python/Tutorial/Advanced/trajectory_io.py. More detailed documentation for log file can be founded here. http://redwood-data.org/indoor/fileformat.html. 

In a long run, it is inconsistent context. Reconstruction system uses .json file to write/optimize odometry, but rgbd_integration example uses .log file. @qianyizh: how do you think?

**Comment by argosvr** [comment]
Thank you for your answer.
I think the .json file format is a very good and easy solution. And very usable before you find a better solution.
I had a look at http://redwood-data.org/indoor/fileformat.html and src/Python/Tutorial/Advanced/trajectory_io.py.
My question is: using http://www.open3d.org/docs/tutorial/Basic/rgbd_odometry.html wich gives me very good results, how can I generate the trajectory log file ?
It would be a great thing to have a piece of code to do that and will bridge a very wide gap for 3d reconstruction..

**Comment by argosvr** [comment]
when I read this: http://redwood-data.org/indoor/fileformat.html
I understand how the metada are working. But to generate the 4-by-4 transformation matrix, what is the command I should use ?

**Comment by syncle** [comment]
Yes. Use that 4x4 transformation matrix for the log file. Depending on the transformation directions (which one is source or target), you will need to apply matrix inverse.

**Comment by argosvr** [comment]
well.. I tried to make a log file using this:
                  with open("navigation.log", "w") as f:
                        f.write(trans_color_term)

it does not work

**Comment by syncle** [comment]
No no. You don't have try to make yourself. As I mentioned earlier, use the following:
https://github.com/IntelVCL/Open3D/blob/master/src/Python/Tutorial/Advanced/trajectory_io.py

**Comment by argosvr** [comment]
well.. this code is just not documented !

**Comment by syncle** [comment]
Try something like below:
```
from trajectory_io import *
import numpy as np

traj = []
for i in range(10):
    transform = np.identity(4)
    metadata = "0 0 0"
    traj.append(CameraPose(metadata, transform))
write_trajectory(traj, "test.log")
traj_read = read_trajectory("test.log")
```
You can replace ``transform`` and ``metadata``.

**Comment by argosvr** [comment]
thanks.. is the transform can be something like trans_color_term ? (using your example)...

**Comment by argosvr** [comment]
great it worked ! thank you

**Comment by d-b-h** [comment]
Hi @syncle  ! I know this issue was closed a while ago, but in "rgbd_integration.py", in this snippet ` volume.integrate(
            rgbd,
            o3d.camera.PinholeCameraIntrinsic(
                o3d.camera.PinholeCameraIntrinsicParameters.PrimeSenseDefault),
            np.linalg.inv(camera_poses[i].pose))`


the intrinsics seem to taken automatically. I am applying the above steps on my own rgbd images with different intrinsics and image height+width. I was able to create trajectory.log but am getting the following error:
Traceback (most recent call last):
  File "rgbd_integration.py", line 30, in <module>
    np.linalg.inv(camera_poses[i].pose))
RuntimeError: [Open3D ERROR] [ScalableTSDFVolume::Integrate] Unsupported image format.


I think this is because I have the wrong instrinsics. Can you help?
